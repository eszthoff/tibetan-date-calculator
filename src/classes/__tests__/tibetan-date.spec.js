import MockDate from 'mockdate';
import TibetanDate from '../tibetan-date';

describe('TibetanDate', () => {
  beforeAll(() => {
    MockDate.set('2019/06/07');
  });
  afterAll(() => {
    MockDate.reset();
  });

  it('should create a correct class when invoked without arg', () => {
    expect(new TibetanDate()).toMatchSnapshot();
  });
  it('should create a correct class when invoked with string arg', () => {
    expect(new TibetanDate('2019/06/22')).toMatchSnapshot();
  });
  it('should create a correct class when invoked with object arg', () => {
    expect(new TibetanDate({
      year: 2146, month: 3, day: 10, isLeapMonth: false, isLeapDay: false
    })).toMatchSnapshot();
  });
});
