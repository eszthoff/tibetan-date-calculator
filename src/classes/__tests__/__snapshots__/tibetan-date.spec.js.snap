// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TibetanDate should create a correct class when invoked with object arg 1`] = `
TibetanDate {
  "date": 10,
  "isDoubledDay": false,
  "isLeapDay": false,
  "isPreviousSkipped": false,
  "isSkippedDay": false,
  "month": 3,
  "monthObj": Object {
    "isDoubledMonth": false,
    "isLeapMonth": false,
    "month": 3,
  },
  "westernDate": 2019-05-14T00:00:00.000Z,
  "year": 2146,
}
`;

exports[`TibetanDate should create a correct class when invoked with string arg 1`] = `
TibetanDate {
  "date": 19,
  "isDoubledDay": false,
  "isLeapDay": false,
  "isPreviousSkipped": false,
  "isSkippedDay": false,
  "month": 4,
  "monthObj": Object {
    "isDoubledMonth": false,
    "isLeapMonth": false,
    "month": 4,
  },
  "westernDate": 2019-06-21T22:00:00.000Z,
  "year": 2146,
}
`;

exports[`TibetanDate should create a correct class when invoked without arg 1`] = `
TibetanDate {
  "date": 3,
  "isDoubledDay": false,
  "isLeapDay": false,
  "isPreviousSkipped": false,
  "isSkippedDay": false,
  "month": 4,
  "monthObj": Object {
    "isDoubledMonth": false,
    "isLeapMonth": false,
    "month": 4,
  },
  "westernDate": 2019-06-06T22:00:00.000Z,
  "year": 2146,
}
`;
